packages/core/src/messages/responses/leaderboard/leaderboard-fetch.ts

```diff
@@ -0,0 +1,83 @@
+import {SdkError} from '../../../errors';
+import {_RankedElement} from '../grpc-response-types';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseNotFound,
+  ResponseFound,
+} from '../response-base';
+
+/**
+ * Parent response type for a leaderboard fetch by rank or by score request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Found}
+ * - {NotFound}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof LeaderboardFetchByScore.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `LeaderboardFetchByScore.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {}
+
+class _Found extends Response {
+  private readonly _elements: _RankedElement[];
+
+  constructor(elements: _RankedElement[]) {
+    super();
+    this._elements = elements;
+  }
+
+  /**
+   * Returns the elements as an array of objects, each containing an `id`, `score`, and `rank` field.
+   * @returns {{id: bigint, score: number, rank: number}[]}
+   */
+  public valueArray(): {id: number; score: number; rank: number}[] {
```

Comment:
I think let's just call this `values` for now; we can add aliases for it if we end up supporting different return types for this in the future.